community/index/src/main/java/org/neo4j/index/internal/gbptree/TreeNodeDynamicSize.java

```diff
@@ -562,6 +572,11 @@ private int middle( PageCursor leftCursor, int insertPos, KEY newKey, VALUE newV
         return middle;
     }
 
+    private int halfSpace()
+    {
+        return (pageSize - HEADER_LENGTH_DYNAMIC) / 2;
```

Comment:
Looks like this could be calculated once and put as a field, although perhaps the compiler does this anyway...